### PR TITLE
Fix travis coverage

### PR DESCRIPTION
**Pull request by tyll**
This change is [Reviewable](https://reviewable.io/reviews/linux-system-roles/network/120)

### CODE REVIEW

File: travis-fix-coverage.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#! /bin/bash
+
+cat > .coveragerc <<EOF
+[paths]
+source =
+    .
+    $PWD
+EOF
+mv .coverage .coverage.merge
```
Comment (i386x) [inline travis-fix-coverage.sh]: Is `.coverage` the right name? `mv` complains that there is no `.coverage`: https://travis-ci.org/linux-system-roles/network/jobs/545422641#L238

File: travis-fix-coverage.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+coverage combine --append .
```
Comment (i386x) [inline travis-fix-coverage.sh]: Is this related to the command one line above? Can this and previous line be written as
```bash
if [[ -f .coverage ]]; then
  mv .coverage .coverage.merge
  coverage combine --append .
fi
```
?

**Comment by pcahyna** [comment]
@tyll there is some error in the Travis logs that does not result in a failure of the test, but still indicates there is something wrong: https://travis-ci.org/linux-system-roles/network/jobs/545422641#L238

File: travis-fix-coverage.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#! /bin/bash
```
Comment (i386x) [inline travis-fix-coverage.sh]: @tyll, is this script necessary as it is not invoked within `tox.ini`?

File: tox.ini [hunk 4]
```diff
@@ -139,15 +139,14 @@ commands =
     flake8 --statistics {posargs} .
 
 [testenv:coveralls]
+changedir = {toxinidir}
 envdir = {toxworkdir}/env-{env:TRAVIS_PYTHON_VERSION:coveralls}
 passenv = TRAVIS TRAVIS_*
-whitelist_externals = bash
```
Comment (i386x) [inline tox.ini]: `whitelist_externals` should be `mv` imho.

File: travis-fix-coverage.sh (hunk 3 above)
Comment (tyll) [reply travis-fix-coverage.sh]: Please ignore this PR for now, it is not ready it I am in the middle of investigating it. A new version of python-coverage was released, that make allow a different approach.

Comment (i386x) [reply travis-fix-coverage.sh]: Ok, thank You. I walked through several commits and found that in the case of d945925 the results was reported successfully (https://coveralls.io/jobs/56225730).

**Comment by i386x** [comment]
If I understand it well, this happens:
1. Travis spawns virtual machine for each job specified in `.travis.yml`
1. on every virtual machine, virtual env is activated for a given version of Python and within it is run `tox`
1. `tox` runs `pytest`, for `py27` and `py36` the coverage results are reported to [coveralls.io](https://coveralls.io/)

The problem here is that recently [coveralls.io](https://coveralls.io/) is not able to combine results from multiple Travis jobs correctly (please, correct me if I am wrong). Some possibilities how to solve this:
- [parallel builds webhook](https://docs.coveralls.io/parallel-build-webhook)
- using [Travis build stages together with a file sharing service](https://docs.travis-ci.com/user/build-stages/#sharing-files-between-jobs-via-s3)
- using [codecov.io](https://codecov.io/)

What's your tips?

Some references:
1. https://github.com/mitodl/mitx-grading-library/pull/226
1. https://github.com/travis-ci/beta-features/issues/42
1. https://github.com/lemurheavy/coveralls-public/issues/1224
1. https://travis-ci.community/t/how-can-i-share-coverage-information-between-build-stages-in-prs-from-forks/859

**Comment by richm** [comment]
@tyll if you still need this PR, go ahead and fix it and merge it into network, and I'll extract the common parts for template

**Comment by tyll** [comment]
So, one problem that I caused IIRC was that I requested coveralls to be included only for 3.5. This was wrong since for the roles it needs to be included for each unit test. I was confusing the setup with nmstate, where coveralls runs in [`after_success`](https://github.com/nmstate/nmstate/blob/master/.travis.yml#L61). Maybe this would also be a better approach here. After fixing this, coverage is reported again but it uses the wrong paths - the coverage is prefixed with `/home/travis/build/...` which makes it annoying to browse through it. I guess this is what I intentionally wanted to fix in this PR.

**Comment by tyll** [comment]
I will merge this to restore basic coverage reporting and then possibly follow-up regarding the path.
